python/fledge/services/core/api/browser.py

```diff
@@ -608,3 +612,52 @@ async def asset_readings_with_bucket_size(request: web.Request) -> web.Response:
         raise web.HTTPInternalServerError(reason=str(e))
     else:
         return web.json_response(response)
+
+
+async def asset_structure(request):
+    """ Browse all the assets for which we have recorded readings and
+    return a readings count.
+
+    Returns:
+           json result on basis of SELECT asset_code, count(*) FROM readings GROUP BY asset_code;
+
+    :Example:
+            curl -sX GET http://localhost:8081/fledge/asset
+    """
+    _logger.error("Strcuture called")
+    payload = PayloadBuilder().GROUP_BY("asset_code").payload()
+
+    results = {}
+    try:
+        _readings = connect.get_readings_async()
+        results = await _readings.query(payload)
+        _logger.error("Strcuture query run")
+        assets = results['rows']
+        asset_json = {}
+        for asset in assets:
+            _logger.error("Processing asset %s", asset['asset_code'])
+            code = asset['asset_code']
+            datapoint = {}
+            metadata = {}
+            _logger.error("Start readings")
+            for name, value in asset['reading'].items():
+                if type(value) == str:
+                    if value == "True" or value == "False":
+                        datapoint[name] = "boolean"
+                    else:
+                        metadata[name] = value
+                elif type(value) == int:
+                    datapoint[name] = "integer"
+                elif type(value) == float:
+                    datapoint[name] = "float"
```

Comment:
if type(value) other than `str, int, float`?

Reply:
Those are the only things that are needed.